integration/testrunner/grpc.go

```diff
@@ -230,10 +270,16 @@ func testGRPCEndpoint(ctx context.Context, t *testing.T, g *GRPC) {
 		} else {
 			t.Errorf("Did not get err as expected. Instead got reply: %v", reply)
 		}
-
-		query := g.makeQueryForGRPCStream(id)
+		want := int64(4)
+		query := g.makeQueryForGRPCStream(id, fieldsNameMap)
+		time.Sleep(10 * time.Second)
+		t.Log(query.Q)
 		// we expect to have 4 audit logs - the last sent number (5) will be after the err occurred.
-		queryIfAuditLogsExistWithRetries(ctx, t, query, g.Config, "stream_fail_on_four", int64(4))
+		value := queryIfAuditLogsExistWithRetries(ctx, t, query, g.Config, "stream_fail_on_four")
+		result := parseQueryResultForGRPCStream(t, value)
+		if diff := cmp.Diff(want, result); diff != "" {
+			t.Errorf(diff)
+		}
```

Comment:
I'm seeing a bunch of repetition across these tests. Could they be table-driven instead?

Reply:
I might be getting this wrong, I think I am following the same format as other test files like [this](https://github.com/abcxyz/lumberjack/blob/861b2c864a2dce43afa50a6c9fdcc97594ae0247/pkg/server/server_test.go#L153-L162).